functions/Test-DbaSqlPath.ps1

```diff
@@ -56,41 +56,53 @@ function Test-DbaSqlPath {
 
 	#>
 	[CmdletBinding()]
-	[OutputType([bool])]
 	param (
 		[Parameter(Mandatory = $true)]
 		[Alias("ServerInstance", "SqlServer")]
-		[DbaInstance[]]$SqlInstance,
+		[DbaInstance]$SqlInstance,
 		[PSCredential]$SqlCredential,
 		[Parameter(Mandatory = $true)]
-		[string]$Path,
+		[string[]]$Path,
 		[switch][Alias('Silent')]$EnableException
 	)
-
+	begin {
+		try {
+			$server = Connect-SqlInstance -SqlInstance $SqlInstance -SqlCredential $SqlCredential
+		}
+		catch {
+			Stop-Function -Message "Failure" -Category ConnectionError -ErrorRecord $_ -Target $SqlInstance -Continue
+		}
+	}
 	process {
-		foreach ($instance in $SqlInstance) {
-			try {
-				$server = Connect-SqlInstance -SqlInstance $instance -SqlCredential $SqlCredential
-			}
-			catch {
-				Stop-Function -Message "Failure" -Category ConnectionError -ErrorRecord $_ -Target $instance -Continue
-			}
-
-			Write-Message -Level VeryVerbose -Message "Checking access to $path for $instance."
-			$sql = "EXEC master.dbo.xp_fileexist '$path'"
-			try {
-				Write-Message -Level Debug -Message "Executing: $sql."
-				$fileExist = $server.Query($sql)
-			}
-
-			catch {
-				Stop-Function -Message "Failed to test the path $Path." -ErrorRecord $_ -Target $instance -Continue
-			}
-			if ($fileExist[0] -eq 1 -or $fileExist[1] -eq 1) {
-				return $true
+		if (Test-FunctionInterrupt) { return }
+		
+		$counter = [pscustomobject] @{ Value = 0 }
+		$groupSize = 100
+		$groups = $Path | Group-Object -Property { [math]::Floor($counter.Value++ / $groupSize) }
+		foreach($g in $groups) {
+			$paths_batch = $g.Group
+			$q = @()
+			foreach($p in $paths_batch) {
+				$q += "EXEC master.dbo.xp_fileexist '$p'"
 			}
-			else {
-				return $false
+			$sql = $q -join ';'
+			$batchresult = $server.ConnectionContext.ExecuteWithResults($sql)
+			if ($Path.Count -eq 1) {
+				if ($batchresult.Tables.rows[0] -eq $true -or $batchresult.Tables.rows[1] -eq $true) {
+					return $true
+				} else {
+					return $false
+				}
+			} else {
+				$i = 0
+				foreach($r in $batchresult.tables.rows) {
+					$doespass = $r[0] -eq $true -or $r[1] -eq $true
+					[pscustomobject]@{
+						filepath = $paths_batch[$i]
+						fileexists = $doespass
```

Comment:
This should have property names cased as `FilePath` and `FileExists`

Also why the underscore in the variable name (`$paths_batch`)?